src/video_core/amdgpu/liverpool.cpp

```diff
@@ -175,6 +175,11 @@ Liverpool::Task Liverpool::ProcessGraphics(std::span<const u32> dcb, std::span<c
     while (!dcb.empty()) {
         const auto* header = reinterpret_cast<const PM4Header*>(dcb.data());
         const u32 type = header->type;
+        if (type == 2) {
```

Comment:
I doubt it is a real type2 packet here. To convince me in opposite, show pm4 stream dump or better gnm function which writes it. We haven't encountered any packets other than type3 so far, and I don't think your case is special. This may be the result of command list corruption, or something else bad is happening.

Reply:
I had the same doubt but the overall command list looks legit for the following reasons:
- Beside that one type-2 packet, every other command list that was submitted has no incoherent data
- The AMD driver in mesa does emit some type-2 packets (look for string `PKT2_NOP_PAD`)
- The type-2 packet itself is just `0x80000000`. So it doesn't look like uninitialized data or whatever

However I didn't go much further to check if this is indeed what the game code does. I could do that but I don't have enough experience with the PS4-specific driver to easily locate the functions that assemble the command list throughout all the code without spending too much time on it. If you happen to be knowledgeable about it, I'd appreciate some pointers.

You can find the command list here as well as a disassembled version:
[00000000_dcb_1_0.zip](https://github.com/user-attachments/files/16736210/00000000_dcb_1_0.zip)

Reply:
To me, this cmd list looks like a part of something bigger, so I assume that type2 is actually the last dword of the previous type3 packet. Dst address of write data also looks suspicious, but it is another thing.
I would recommend excluding this change from the PR, so we will be able to merge it. This can be addressed as a separate fix, however more investigations are needed. Also, please specify what game you're testing, as it will be helpful in tracking such issues.

Reply:
Sure, I'll move the type-2 specific changes to another PR with and will add more info there.

Reply:
Done.